iree/compiler/InputConversion/MHLO/test/convert_structural_types.mlir

```diff
@@ -0,0 +1,26 @@
+// RUN: iree-opt -split-input-file --iree-mhlo-to-linalg-on-tensors %s | IreeFileCheck %s
+
+// CHECK-LABEL: @func_cfg_conversion
+module @func_cfg_conversion {
+  // CHECK: func @caller(%arg0: tensor<2xi32>, %arg1: i1) -> tensor<1xi32>
+  func @caller(%arg0: tensor<2xui32>, %arg1 : i1) -> tensor<1xui32> {
+    // CHECK: call @callee(%arg0, %arg1) : (tensor<2xi32>, i1) -> tensor<2xi32>
+    %0 = call @callee(%arg0, %arg1) : (tensor<2xui32>, i1) -> tensor<2xui32>
+    %1 = "mhlo.slice"(%0) {limit_indices = dense<1> : tensor<1xi64>, start_indices = dense<0> : tensor<1xi64>, strides = dense<1> : tensor<1xi64>} : (tensor<2xui32>) -> tensor<1xui32>
+    // CHECK: return %1 : tensor<1xi32>
+    return %1 : tensor<1xui32>
+  }
+
+  // CHECK: func @callee(%arg0: tensor<2xi32>, %arg1: i1) -> tensor<2xi32>
+  func @callee(%arg0: tensor<2xui32>, %arg1: i1) -> tensor<2xui32> {
+    // CHECK: cond_br %arg1, ^bb1(%arg0 : tensor<2xi32>), ^bb2(%arg0 : tensor<2xi32>)
+    cond_br %arg1, ^bb1(%arg0 : tensor<2xui32>), ^bb2(%arg0 : tensor<2xui32>)
+  // CHECK: ^bb1(%0: tensor<2xi32>)
```

Comment:
These tests seem pretty brittle, especially where the SSA value names actually change

Reply:
Thanks for the catch. Had forgotten to generalize this test.